lib/manageiq/release/internationalization.rb

```diff
@@ -0,0 +1,103 @@
+module ManageIQ
+  module Release
+    class Internationalization
+
+      attr_reader :branch, :dry_run
+
+      def initialize(branch, dry_run: true)
+        @branch  = branch || 'master'
+        @dry_run = dry_run
+      end
+
+      def execute(command, description = nil)
+        description ||= "*** Running `#{command}` ***"
+        puts description
+        system(command)
+      end
+
+      def create_database_yml
+        puts "*** Creating config/database.yml ***"
+        database_config = YAML.load File.read("config/database.pg.yml")
+        unless database_config.key?('i18n')
+          database_config['i18n'] = database_config['test'].dup
+          database_config['i18n']['database'] = "vmdb_i18n"
+        end
+        File.open("config/database.yml", "w") { |file| file.write(database_config.to_yaml) }
+      end
+
+      def manageiq_pot_filename
+        "locale/manageiq.pot"
+      end
+
+      def manageiq_pot_file_needs_commit?
+        needs_commit = false
+        diffs = git_diff_stats
+        return false if diffs.length != 1
+        return false unless diffs.key?(manageiq_pot_filename)
+        stats = diffs[manageiq_pot_filename]
+        puts "+#{stats[:insertions]}, -#{stats[:deletions]} for #{manageiq_pot_filename}"
+          # If the only changes are to POT-Creation-Date and PO-Revision-Date,
+          # then there are no substantive changes
+          # Hence why this code checks for precisely 2 insertions and deletions
+        (stats[:insertions] != 2) || (stats[:deletions] != 2)
+      end
+
+      def git_diff_stats
+        MiniGit::Capturing.diff(:numstat => true).split("\n").each_with_object({}) do |diff, stats|
+          insertions, deletions, fname = diff.split("\s")
+          stats[fname] = {:insertions => insertions.to_i, :deletions => deletions.to_i }
+        end
+      end
+
+      def manageiq_repo_path
+        manageiq_repo.path || File.join(__dir__, "../../../repos/ManageIQ/manageiq")
+      end
+
+      def manageiq_repo
```

Comment:
Naming wise, prefer `repo`, since that's consistent with other classes.

Reply:
There will also be classic ui and service ui repos here later.  Hence the name `manageiq_repo`

Reply:
I have to wrap my head around that since usually the way it works is you write a class working with a single repo, then just iterate over the repos.  Ideally I'd expect each repo would have a similar "interface" for updating the locales.

Reply:
would that it were so simple. ;-)  let's flesh out the whole thing and then we can refactor and rename things.

Reply:
refactored into classes